storage/rocksdb/rdb_datadic.cc

```diff
@@ -780,6 +783,53 @@ int Rdb_key_def::successor(uchar *const packed_tuple, const uint &len) {
   return changed;
 }
 
+bool Rdb_key_def::covers_lookup(TABLE *const table,
+                                const rocksdb::Slice *const unpack_info) const {
+  DBUG_ASSERT(m_index_type == INDEX_TYPE_SECONDARY);
+  DBUG_ASSERT(table->key_info != nullptr);
+  Rdb_string_reader unp_reader = Rdb_string_reader::read_or_empty(unpack_info);
+
+  // Check if unpack_info is corrupted
+  const bool has_unpack_info =
+      unp_reader.remaining_bytes() &&
+      *unp_reader.get_current_ptr() == RDB_UNPACK_DATA_TAG;
+  if (!has_unpack_info || !unp_reader.read(RDB_UNPACK_HEADER_SIZE)) {
+    return false;
+  }
+
+  MY_BITMAP covered_bitmap;
+  my_bitmap_map covered_bits;
+  bitmap_init(&covered_bitmap, &covered_bits, 16, false);
+  if (unp_reader.read_uint16(&covered_bits)) {
+    return false;
+  }
+
+  uint covered_columns = 0;
+  KEY key_info = table->key_info[get_keyno()];
```

Comment:
ha_rocksdb::open() is called to construct the handler to the table used for the scan. index_init() is invoked to setup_read_decoders(), so this might be the place for it. You can cache the calculation in ha_rocksdb. Just be sure it gets cleared appropriately before the structure is reused.

Also verify it works with the index_merge query plan. There are some test cases that generate index_merge query plans, but none which use varchar() keys, so it'll be good to ensure index_merge query plans work well with the caching scheme because the query plan also opens up the table for scanning and performs the initialization differently from the regular secondary index scan.